netty/src/main/java/io/grpc/netty/NettyHandlerSettings.java

```diff
@@ -56,17 +63,23 @@ public static synchronized void autoWindowOn(boolean autoFlowControl) {
     autoFlowControlOn = autoFlowControl;
   }
 
-  public static synchronized int getLatestClientWindow() {
-    return getLatestWindow(clientHandler);
+  public static synchronized Future<Integer> getLatestClientWindow() {
+    return getLatestWindow(clientLoop, clientHandler);
   }
 
-  public static synchronized int getLatestServerWindow() {
-    return getLatestWindow(serverHandler);
+  public static synchronized Future<Integer> getLatestServerWindow() {
+    return getLatestWindow(serverLoop, serverHandler);
   }
 
-  private static synchronized int getLatestWindow(AbstractNettyHandler handler) {
+  private static synchronized Future<Integer> getLatestWindow(
```

Comment:
As this is run in the loop, `synchronized` seems not necessary.

Reply:
I guess this method doesn't need it, but the calls do.   The need to synchronize getting the loop and the handler with `setAutoWindow`